File: tasks/whoseTurn.js
function whoseTurn(p) {

  var color = function(r, c) {
    return(r + c) % 2;
  }

  var s = 0;
  for(var i = 0; i < 4; i++) {
    var c = p.charCodeAt(i * 3 + 0) - 'a'.charCodeAt();
    var r = p.charCodeAt(i * 3 + 1) - '0'.charCodeAt();
    s += color(r, c);
  }

  return s % 2 === 0;
}

File: tasks/whoseTurn.java
boolean whoseTurn(String p) {

  class Helper {

  int color(int r, int c) {
    return (r + c) % 2;
  }
  }
  Helper h = new Helper();

  int s = 0;
  for (int i = 0; i < 4; i++) {
    int c = p.charAt(i * 3 + 0) - 'a';
    int r = p.charAt(i * 3 + 1) - '0';
    s += h.color(r, c);
  }

  return s % 2 == 0;
}
